quinn-proto/src/endpoint.rs

```diff
@@ -278,6 +268,17 @@ impl Endpoint {
         // If we got this far, we're a server receiving a seemingly valid packet for an unknown
         // connection. Send a stateless reset if possible.
         //
+
+        if !first_decode.is_initial()
```

Comment:
Is there a reason not to introduce it in this spot in the first place? This move seems somewhat independent of using the keyed generator by default.